src/main/java/com/hyunec/cosmicbaseballinit/vo/HitterResult.java

```diff
@@ -1,10 +1,56 @@
 package com.hyunec.cosmicbaseballinit.vo;
 
-
 import lombok.RequiredArgsConstructor;
 
 @RequiredArgsConstructor
 public enum HitterResult {
-    GO(),
-    OUT();
+    STRIKE_OUT(null, null),
+    FOUR_BALL(null, null),
+    HIT(null, null),
+    BULLSEYE_STRIKE(0.2, PitchResult.STRIKE),
+    BULLSEYE_BALL(0.2, PitchResult.BALL),
+    HOMERUN(0.2, PitchResult.HIT);
+
+    private final Double probability;
+    private final PitchResult pitchResult;
+
+    public static HitterResult judgeHitterResultByPitchResult(PitchResult lastPitchResult, Integer count)
+            throws Exception {
+        if (lastPitchResult.getValue() == count) {
+            return judgeInCaseOfNotSpecial(lastPitchResult);
+        }
+        return judgeInCaseOfSpecial(lastPitchResult, Math.random());
+    }
+
+    public static HitterResult judgeInCaseOfSpecial(PitchResult pitchResult,
+                                                     Double randomDouble) throws Exception {
+        HitterResult hitterResult = findHitterResultByPitchResult(pitchResult);
+        if (randomDouble < hitterResult.probability) {
+            return hitterResult;
+        }
+        return null;
+    }
+
+    public static HitterResult judgeInCaseOfNotSpecial(PitchResult pitchResult) throws Exception {
+        if (pitchResult == PitchResult.STRIKE) {
+            return STRIKE_OUT;
+        }
+        if (pitchResult == PitchResult.BALL) {
+            return FOUR_BALL;
+        }
+        if (pitchResult == PitchResult.HIT) {
+            return HIT;
+        }
+        throw new Exception("HitterResult error");
+    }
+
+    private static HitterResult findHitterResultByPitchResult(PitchResult pitchResult) throws Exception {
+        for (HitterResult hitterResult : HitterResult.values()) {
+            if (hitterResult.pitchResult == null) continue;
+            if (hitterResult.pitchResult.equals(pitchResult)) {
+                return hitterResult;
+            }
+        }
+        throw new Exception("findHitterResultByPitchResult error");
```

Comment:
`nit` 
continue 활용은 꽤 위험한 방법입니다.
stream 을 활용하면 조금 더 깔끔한 코딩이 가능할 것 같습니다.

Reply:
stream에 익숙하지 않아서 잘 사용하지 않고 있었는데, stream 활용법을 익혀보겠습니다. 인터넷을 찾아보니 스트림에 관한 글들은 주로 effective 자바 라는 책을 인용하는 것을 느껴서, 해당 책을 구매하려고 합니다. 괜찮은 선택일까요?

Reply:
네. 이펙티브 자바는 꼭 구입하시어 1년에 한번씩 읽어보시기를 추천합니다.

Reply:
```suggestion
return Arrays.stream(HitterResult.values())
                .filter(hitterResult -> hitterResult.pitchResult.equals(pitchResult))
                .findFirst()
                .orElseThrow(() -> new Exception("findHitterResultByPitchResult error"));
```

스트림으로 정리하면 위처럼 될 수 있을 것 같습니다.